enhancements/authentication/separate-oauth-resources.md

```diff
@@ -0,0 +1,142 @@
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-10-16
+last-updated: 2019-10-16
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Separate OAuth API-Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+The API resources used by the OpenShift OAuth server will be moved to their own `openshift-oauth-apiserver` repository.
+The openshift-oauth-apiserver binary will be managed by the existing authentication-operator.
+
+## Motivation
+
+The OpenShift OAuth server requires several openshift API resources in order to function.
+Because the OAuth server enables access to the kube-apiserver, it needs to be able to run separately from the existing
+set of API resources that are focused on enabling a developer workflow.
+This narrowing of scope expands the deployment options for the OpenShift OAuth server.
+
+### Goals
+
+1. Make it possible to run the OpenShift OAuth server without installing all the developer focused API Resources.
+
+### Non-Goals
+
+1. Move routes out of the OpenShift API Server.
+
+## Proposal
+
+1. Create a new repo called `openshift-oauth-apiserver` that produces a binary called `openshift-oauth-apiserver`.
+2. Add the following API groups to the new repo
+   1. authorization  - most of this API group is deprecated.  It will sleep here until we're ready to retire it.
+   2. oauth - core oauth types
+   3. project - intersection of authorization and namespaces.  It is heavily dependent on the way our scopes work.
+   4. quota - intersection of users and namespaces
+   5. security - SCC
+   6. user - core types used by oauth and authentication
+3. Update the authentication operator to install the new apiserver, but not wire the apiservice.
+4. Add the code to the openshift-apiserver-operator tolerate apiservice management by the authentication operator
+(indicated by annotation) if the authentication operator is available at the 4.n version.  This code will need to be 
+backported to 4.n-1.
+5. Update the authentication operator to wire the apiservices.
+6. Remove the registration from the openshift-apiserver-operator.
+7. Remove the serving code from the openshift-apiserver.
+
+If we stop at any any intermediate step, we are still safe to ship.
+
+### User Stories [optional]
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they releate.
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
```

Comment:
Like to see operational cost increases here

Reply:
> Like to see operational cost increases here

Sure, I'll fill in something like: this will mirror one of the most reliable operators in the field openshift-apiserver.  It won't actually be another operator, it will be added to the cluster-authentication-operator.